### PR TITLE
dev-lang/gnucobol: rename from dev-lang/open-cobol

### PR DESCRIPTION
**Pull request by XVilka**
Because mainstream changed the project name starting
from the 2.0 versions, along with license (GPL3/LGPL3).
Also bumps to the latest stable 2.2 version.

Package-Manager: Portage-2.3.66, Repoman-2.3.12
Closes: https://bugs.gentoo.org/641888

### CODE REVIEW

**Comment by XVilka** [comment]
Ping?

**Comment by XVilka** [comment]
Anyone alive here?

**Comment by leio** [comment]
Missing pkgmove entry to have users seamlessly update to the new name. Is the removal of the old version immediately intentional?

**Comment by XVilka** [comment]
@leio, I wasn't sure, I can keep the old version if required. Will update the PR.

**Comment by leio** [comment]
I don't know, that's for the maintainer to decide, I suppose. I think if removed, it should be a separate commit, though easier to do it in the same, as you wouldn't need to adjust the old ebuild to work under the new name.

**Comment by XVilka** [comment]
Ok, canceled removal updated the ebuild and added also gnucobol-3.0-rc1 version.

**Comment by XVilka** [comment]
Ping?

**Comment by mgorny** [comment]
I'm sorry but it seems that the current maintainer doesn't care about this package. Are you still around? If so, would you be interested in becoming a new maintainer? If so, please replace the current maintainer with yourself in metadata.xml as found in https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml

**Comment by XVilka** [comment]
Yep, will do in a couple days. Thanks!

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MY_PV=$(ver_cut 1-2)
+MY_P="${PN}-${MY_PV}-rc1"
```
Comment (mgorny) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild]: `MY_P=${P/_/-}` is how we usually deal with it. Especially, don't hardcode `rc1`.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+DESCRIPTION="an open-source COBOL compiler"
+HOMEPAGE="https://open-cobol.sourceforge.io"
+SRC_URI="mirror://sourceforge/open-cobol/${MY_P}.tar.gz"
+
+LICENSE="GPL-3 LGPL-3"
```
Comment (mgorny) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild]: The sources say it's 'or newer', i.e. `GPL-3+ LGPL-3+`.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="berkdb nls readline"
+
+RDEPEND="dev-libs/gmp:0
```
Comment (mgorny) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild]: Prefer sorting unconditional deps first, then all conditionals. It makes it easier to verify them later.

**Comment by flappyports** [comment]
@XVilka, package has been converted to EAPI=7 and future proofed for pkgmove if you so choose to keep working this PR.

**Comment by XVilka** [comment]
Yes, I will, sorry for the delay. Will update the PR later today.

**Comment by XVilka** [comment]
Addressed everything. Please recheck.

**Comment by XVilka** [comment]
Anything else that I need to do? Lets merge it finally if possible @leio @mgorny

**Comment by XVilka** [comment]
Ping? Anything else I should do?

**Comment by XVilka** [comment]
Ping?

**Comment by XVilka** [comment]
Updated the pkgmove to 1Q-2020, please review.

**Comment by XVilka** [comment]
Ping?

**Comment by XVilka** [comment]
With the renewed interest in COBOL (it's even [trending on Twitter](https://news.ycombinator.com/item?id=22784248)) after the famous [New Jersey call for COBOL developers](https://www.cnn.com/2020/04/08/business/coronavirus-cobol-programmers-new-jersey-trnd/index.html) to fix their systems and many people trying the free solutions like GnuCOBOL, please merge it as soon as possible - so these people will get the most relevant experience (a lot has changed since the rename!). This is the last call, thanks. Otherwise I just close this.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2019 Gentoo Authors
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:1]: Date

File: dev-lang/gnucobol/gnucobol-2.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Copyright 1999-2019 Gentoo Authors
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-2.2.ebuild:1]: Date

File: dev-lang/gnucobol/gnucobol-2.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+EAPI=7
+
+DESCRIPTION="an open-source COBOL compiler"
+HOMEPAGE="https://open-cobol.sourceforge.io"
+SRC_URI="mirror://sourceforge/open-cobol/${P}.tar.gz"
+
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="berkdb nls readline"
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-2.2.ebuild:13]: Where does readline get used?

File: dev-lang/gnucobol/gnucobol-2.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+RDEPEND="dev-libs/gmp:0
+	sys-libs/ncurses:0
+	berkdb? ( sys-libs/db:4.8 )"
+DEPEND="${RDEPEND}
+	sys-devel/libtool"
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-2.2.ebuild:19]: Quoting from https://sourceforge.net/projects/open-cobol/files/: 
   ***
    NOTE - libltdl is NOT needed when installing on Linux,
    SUN Solaris, MAC OS, CentOS or Windows
        (including Cygwin, MingW and native windows).
    It is also NOT needed with later versions of AIX and HP-UX.
    (AIX >= 5.1 and HP-UX >= 11.1 are known to NOT require this).
    (Check if you have the "dlopen" function).
    ***

So is libtool needed or not?

File: dev-lang/gnucobol/gnucobol-2.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${P}-disable-data-hooks.patch"
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-2.2.ebuild:22]: ${PN}

**Comment by thesamesam** [comment]
I've spoken to the maintainer (Patrick) and he has given proxy-maintainers permission to move forward with this.

Sorry for the delay, it is not the biggest of projects, and where a package has a maintainer listed, it is not usually priority and can get lost.

I have given a few comments here and it is on the list for the project to look at. Thanks for your patience.

File: dev-lang/gnucobol/gnucobol-2.2.ebuild [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+EAPI=7
+
+DESCRIPTION="an open-source COBOL compiler"
+HOMEPAGE="https://open-cobol.sourceforge.io"
+SRC_URI="mirror://sourceforge/open-cobol/${P}.tar.gz"
+
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-2.2.ebuild:12]: I am suspicious about retaining ppc [keywords](https://devmanual.gentoo.org/keywording/index.html) given how major of a change this is.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+SRC_URI="mirror://sourceforge/open-cobol/${MY_P}.tar.gz"
+
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="berkdb nls readline"
+
+RDEPEND="dev-libs/gmp:0
+	sys-libs/ncurses:0
+	berkdb? ( sys-libs/db:4.8 )"
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:19]: Conditional runtime dependencies aren't allowed, and it looks like this is used at build-time anyway, so BDEPEND or (likely here) DEPEND.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+DEPEND="${RDEPEND}
+	sys-devel/libtool"
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:21]: See comment about libtool above.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild (hunk 10 above)
Comment (XVilka) [reply dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:19]: I took these lines from the open-cobol ebuild: https://github.com/gentoo/gentoo/blob/master/dev-lang/open-cobol/open-cobol-1.1.ebuild#L20

Comment (thesamesam) [reply dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:19]: Don't trust past ebuilds ;)

File: dev-lang/gnucobol/gnucobol-2.2.ebuild [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+EAPI=7
+
+DESCRIPTION="an open-source COBOL compiler"
+HOMEPAGE="https://open-cobol.sourceforge.io"
+SRC_URI="mirror://sourceforge/open-cobol/${P}.tar.gz"
+
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="berkdb nls"
```
Comment (thesamesam) [inline dev-lang/gnucobol/gnucobol-2.2.ebuild:13]: We may need to pull in libntl or gettext if nls is enabled?

Comment (XVilka) [reply dev-lang/gnucobol/gnucobol-2.2.ebuild:13]: Added gettext - looks it is used by the source code.

Comment (thesamesam) [reply dev-lang/gnucobol/gnucobol-2.2.ebuild:13]: In the past, I have done:
BDEPEND="sys-devel/gettext"
DEPEND="virtual/libintl" (to allow for Musl where we need to pull in a package)

But I had to do some investigation to figure out what was needed and when. If you can't figure it out for now, we will leave it.

**Comment by XVilka** [comment]
Let's leave it, they are going to prepare release in a few months, then it
can be reevaluated.

**Comment by XVilka** [comment]
So? Isn't it a good time to merge?

**Comment by XVilka** [comment]
Ping? I addressed feedback multiple times I was asked to. Please either merge as is or just remove the old package completely - it's older than trilobites.

**Comment by thesamesam** [comment]
This is on the proxy-maint proj's list. I can't merge it myself yet (no perms) but it is to be looked at. Please be patient, I understand you've waited a while, but we've had the ACK and will move forward when we can. It won't be anywhere near as long as you already waited.

File: profiles/updates/1Q-2020 [hunk 13]
```diff
@@ -12,3 +12,4 @@ move www-servers/automx2 net-mail/automx2
 move dev-python/pysnmp-apps net-analyzer/snmpclitools
 move net-im/slack-bin net-im/slack
 slotmove sys-libs/libunwind 7 0
+move dev-lang/open-cobol dev-lang/gnucobol
```
Comment (juippis) [inline profiles/updates/1Q-2020:15]: This should be in 2Q-2020 file.

File: dev-lang/gnucobol/metadata.xml [hunk 14]
```diff
@@ -2,8 +2,12 @@
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
 	<maintainer type="person">
-		<email>patrick@gentoo.org</email>
-		<name>Patrick Lauer</name>
```
Comment (juippis) [inline dev-lang/gnucobol/metadata.xml]: Leave him also here, he's on devaway but might have personal interest towards this package when he returns.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 15]
```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MY_P="${P/_/-}"
+
+DESCRIPTION="an open-source COBOL compiler"
```
Comment (juippis) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:8]: `An`

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="berkdb nls"
+
+RDEPEND="dev-libs/gmp:0
+	sys-libs/ncurses:0"
+DEPEND="${RDEPEND}
+	berkdb? ( sys-libs/db:4.8 )
+	nls? ( sys-devel/gettext )"
```
Comment (juippis) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:21]: This is a BDEPEND with EAPI-7.

File: dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="berkdb nls"
+
+RDEPEND="dev-libs/gmp:0
+	sys-libs/ncurses:0"
+DEPEND="${RDEPEND}
+	berkdb? ( sys-libs/db:4.8 )
```
Comment (juippis) [inline dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:20]: Are you sure it doesn't work with later ones? Upstream files don't indicate that anywhere. Also looking at my build.log it seems to be linking against 6.0 even though 4.8.30-r3 is installed. So do something like `>=sys-libs/db-4.1:*` here.

**Comment by juippis** [comment]
The problem here is that the current maintainer is away, and has been for a long time. No one else sees notifications for these bugs / pull requests, and everyone else is pretty much busy with their own projects. Usually people are not that keen on touching packages belonging to others.

**Comment by XVilka** [comment]
Fixed everything except GCC-10. I will check this one later, since the need to install it first. There wasn't any GCC-10 when I opened the pull request...

**Comment by ajakk** [comment]
> Fixed everything except GCC-10. I will check this one later, since the need to install it first. There wasn't any GCC-10 when I opened the pull request...

Did you ever check into this?

**Comment by XVilka** [comment]
@ajakk sorry, had no time on this and will not in the near future (I am busy with other FOSS projects).

**Comment by XVilka** [comment]
If anyone is going to continue to work on this - please note that it's better to package 3.1.2 instead:

GnuCOBOL 3.1.2 — Latest release
Released 23 December 2020

GnuCOBOL 3.1.2 includes many new features compared to the previous release, while maintaining full source compatibility - (too much to note here but found in the linked NEWS), here are some of the highlights:

Huge improvements for compatibility to different COBOL dialects, better error handling and adjustable exceptions per COBOL 2002/2014; more modern format for diagnostic messages (especially useful when used in an integrated development environment possible in Emacs, Vim, VSCodium and others) and improved source-level debugging.

https://sourceforge.net/projects/gnucobol/files/gnucobol/3.1/gnucobol-3.1.2.tar.xz/download

**Comment by atupone** [comment]
rename done

**Comment by thesamesam** [comment]
Thank you @tupone and thank you @XVilka, I'm sorry this took so long.

**Pull request closed without merging**